model/feature_flags.go

```diff
@@ -20,13 +20,17 @@ type FeatureFlags struct {
 	CollapsedThreads bool
 	// Feature flags to control plugin versions
 	PluginIncidentManagement string `plugin_id:"com.mattermost.plugin-incident-management"`
+
+	// AppsEnabled toggle the Apps framework functionalities both in server and client side
+	AppsEnabled bool
```

Comment:
The intended use of feature flags for plugins is to use the functionally like Incident is doing above. Where it will install a specified version from the marketplace and automatically enable it. This way you can roll out new version of the plugin, not just enable and disable it. 
It also gives the user control to enable/disable the plugin and prevent automatic upgrade from taking place.

Reply:
@catalintomai @levb Is this what we want?
@crspeller We might not want to allow the user to disable / enable this plugin. It is an open discussion right now.

Reply:
@crspeller after internal discussion, we have thought the following:
- At the moment, we are not interested in using the features flags the way Incident Management uses them. We are not exactly forcing a version.
- We don't want users to be able to enable or disable the plugin. We will also add the needed steps so the plugin becomes a "special" plugin like NPS. The reason for this is that even if the apps framework is implemented as a plugin for isolation, we prefer it being considered as part of the product. Therefore it can only be turned off by feature flags.

Does this make sense?

Reply:
@larkox Sounds like you just want a configuration setting then. Or maybe just a special environment variable?  Feature flags are for rolling out new features or plugin versions, then are designed to be removed. This sounds more perminant?
When do you see someone using this flag?

Reply:
We plan to have this feature flag until we are confident with the feature. When we are confident with the results of the feature, we can remove the feature flag. We may add more tooling to keep this "always enabled" functionality in the future, but right now this is the plan, at least the way I see it.
@catalintomai @levb Any comment?

Reply:
Presumably you plan to do this more than once? What are you going to do when you have 1.0 of your plugin and you want to roll out 1.1?

Reply:
We can do smth. similar - add a different feature flag for the new functionality and remove it when we are confident that the functionality is ok. Is it mandatory to reuse an existing feature flag for a particular area of the product (this plugin in this case)?

Reply:
> Is it mandatory to reuse an initial/existing feature flag for a particular area of the product (this plugin in this case)?

No

> the way we have things setup now - it triggers a version change

Ya. It's less granular then the feature flags can be on the server. But since plugins generally have less per release and are managed by versions, it makes sense to flag by version. 

> and since we don't have them yet, then we'll have to use a feature flag

Feature flags are kill switches. 

> or should we use a new one for that purpose (and not tie it to the version one)

You can. if you want more granular control, in addition to version control, you can use additional feature flags to control specific features on the plugin. The feature flags come in though the config to plugins.

Reply:
I made another Push. As a summary:
- We want apps to behave as part of the system. Therefore, we want it to be always active.
- We want the version enforcing and the kill switch.
- Version enforcing will allow us to make sure all instances are at the version we want.
- Kill switch will allow us to turn off the apps if necessary.

Does this make sense @crspeller ? @catalintomai , am I mising anything?